### PR TITLE
Support different learning rate scheduler for param group

### PR DESCRIPTION
**Pull request by emailweixu**
This is used for S5 training.

### CODE REVIEW

File: alf/optimizers/optimizers.py [hunk 1]
```diff
@@ -198,8 +199,14 @@ def step(self, closure=None):
         """
         if self._lr_scheduler is not None:
             lr = float(self._lr_scheduler())
-            for param_group in self.param_groups:
-                param_group['lr'] = lr
+            for i, (lr_scheduler, param_group) in enumerate(
+                    zip(self._lr_schedulers, self.param_groups)):
+                if lr_scheduler is not None:
+                    param_group['lr'] = lr_scheduler()
```
Comment (breakds) [inline alf/optimizers/optimizers.py:205]: If my understanding is correct, in this particular case `param_group['lr']` has already been assigned to this scheduler below at line 340 when the group is added?

Comment (emailweixu) [reply alf/optimizers/optimizers.py:205]: 340 is only executed once when the param_group is added.
Here we need to update 'lr' according to lr_schedule for every call to step().

Comment (breakds) [reply alf/optimizers/optimizers.py:205]: Ah I see. This is calling the `scheduler` to acquire the current learning rate. Sorry about the misunderstanding.

**Review by breakds — APPROVED** [review]
